### ISSUE TITLE
Add a LICENSE file

### ISSUE THREAD

**Issue by peterthejohnston** [microsoft/windows-rs#1587]
Hi, would it be possible to add a LICENSE file(s) to the repository? IIUC from the various crates' manifests, e.g. for [windows-sys](https://github.com/microsoft/windows-rs/blob/master/crates/libs/sys/Cargo.toml#L7), the crates are uniformly licensed under "MIT OR Apache-2.0". It would be great to have either a project-wide LICENSE file and/or crate-specific ones to make that more clear.

**Comment by riverar** [comment]
All the crates here are dual licensed for maximum compatibility with the Rust ecosystem. The Apache 2.0 license provides patent cover and MIT provides GPLv2 compatibility.

GitHub doesn't support dual-licensing so I'm a bit worried about how a LICENSE file would impact the UX there. But perhaps we can instead add `LICENSE.Apache-2.0` and `LICENSE.MIT` files.

Will defer to @kennykerr and @jeffwilcox for final Microsoft OSS repository guidance.

**Comment by kennykerr** [comment]
The license files can be found here: https://github.com/microsoft/windows-rs/tree/master/.github

**Comment by peterthejohnston** [comment]
Thanks for pointing those out! I don't think those license files are included in the crates published to crates.io. For example: https://docs.rs/crate/windows/latest/source/

Would it be possible to include them with the published crates?

**Comment by kennykerr** [comment]
Just so I understand, what would be the point of including the files?

**Comment by peterthejohnston** [comment]
Mainly because it allows for vendoring these crates in another open source project (in my case, https://fuchsia.dev/). The Apache-2.0 and MIT License terms require that the license text is included with any redistributed sources, and unless the license-* files are included in the crate published to crates.io, [`cargo vendor`](https://doc.rust-lang.org/cargo/commands/cargo-vendor.html) (which we use to vendor crates) won't pull down the license-* files.

**Comment by kennykerr** [comment]
Got it, well I think I can make that happen.

**Comment by kennykerr** [comment]
Hi Peter, sorry I've been swamped with other work. 

It's a little more work than I expected. I was hoping to drop this into the toml file for the `windows` and `windows-sys` crates:

```
include = ["/src", "../../../.github/license-apache", "../../../.github/license-mit"]
```

But it seems to require the files to be within the crate directory so I'd have to add a build step that copies them there first.

**Comment by kennykerr** [comment]
This should do the trick: https://github.com/microsoft/windows-rs/pull/1602

**Comment by kennykerr** [comment]
This is now published as part of [0.34.0](https://github.com/microsoft/windows-rs/releases/tag/0.34.0).

**Comment by peterthejohnston** [comment]
Excellent, thanks so much!
